RealmSwift/MutableSet.swift

```diff
@@ -471,6 +474,14 @@ extension MutableSet: RealmCollection {
                 convenience should not be relied on.
      */
     public subscript(position: Int) -> Element {
+        if lastAccessedNames != nil {
+            let value = Element._rlmDefaultValue(true)
+            if let value = value as? ObjectBase {
+                value.lastAccessedNames = lastAccessedNames
+                value.prepareForRecording()
+            }
```

Comment:
This logic is repeated in a whole bunch of places and should be extracted to one place. Probably just `Element._rlmKeypathRecorder()` or something?